CMakeLists.txt

```diff
@@ -169,30 +165,18 @@ endif()
 # CUDA support
 #--------------
 
-if(NOT ARB_GPU_MODEL MATCHES "none")
+if(ARB_WITH_GPU)
     set(ARB_WITH_CUDA TRUE)
 
     add_compile_options(
         "$<$<COMPILE_LANGUAGE:CUDA>:-Xcudafe=--diag_suppress=integer_sign_change>"
         "$<$<COMPILE_LANGUAGE:CUDA>:-Xcudafe=--diag_suppress=unsigned_compare_with_zero>")
     target_compile_definitions(arbor-private-deps INTERFACE ARB_HAVE_GPU)
 
-    if(ARB_GPU_MODEL MATCHES "K20")
-        set(cuda_arch sm_35)
-        set(arb_cuda_arch 350)
-    elseif(ARB_GPU_MODEL MATCHES "K80")
-        set(cuda_arch sm_37)
-        set(arb_cuda_arch 370)
-    elseif(ARB_GPU_MODEL MATCHES "P100")
-        set(cuda_arch sm_60)
-        set(arb_cuda_arch 600)
-    else()
-        message(FATAL_ERROR "-- GPU architecture '${ARB_GPU_MODEL}' not supported. Use one of {none, K20, K80, P100}")
-    endif()
+    add_compile_options($<$<COMPILE_LANGUAGE:CUDA>:-gencode=arch=compute_35,code=sm_35>)
+    add_compile_options($<$<COMPILE_LANGUAGE:CUDA>:-gencode=arch=compute_37,code=sm_37>)
+    add_compile_options($<$<COMPILE_LANGUAGE:CUDA>:-gencode=arch=compute_60,code=sm_60>)
```

Comment:
This will apply the options to every nvcc invocation, where we actually just need to compile for all these targets on the library: the unit tests should be fine using whatever the defaults are. I'll edit this one in-place to change these to `target_compile_options`.